src/integrationTest/java/org/opensearch/security/DlsIntegrationTests.java

```diff
@@ -605,6 +647,133 @@ public void testAggregateAndComputeStarRatings() throws IOException {
+        multiGetRequest.add(new MultiGetRequest.Item(FIRST_INDEX_NAME, FIRST_INDEX_ID_SONG_2));
+
+        try (RestHighLevelClient restHighLevelClient = cluster.getRestHighLevelClient(USER_MATCH_ARTIST_BOOL_QUERY)) {
+            assertProperMultiGetResponseForAccessibleAndNonAccessibleDocuments(restHighLevelClient, multiGetRequest);
+        }
+
+        try (RestHighLevelClient restHighLevelClient = cluster.getRestHighLevelClient(USER_MATCH_STARS_TERM_QUERY)) {
+            assertProperMultiGetResponseForAccessibleAndNonAccessibleDocuments(restHighLevelClient, multiGetRequest);
+        }
+    }
+
+    private void assertProperMultiGetResponseForAccessibleAndNonAccessibleDocuments(
+        RestHighLevelClient restHighLevelClient,
+        MultiGetRequest multiGetRequest
+    ) throws IOException, Exception {
+        MultiGetResponse multiGetResponse = restHighLevelClient.mget(multiGetRequest, DEFAULT);
+        List<GetResponse> getResponses = Arrays.stream(multiGetResponse.getResponses())
+            .map(MultiGetItemResponse::getResponse)
+            .collect(Collectors.toList());
+        assertThat(getResponses, hasItem(containDocument(FIRST_INDEX_NAME, FIRST_INDEX_ID_SONG_1)));
+        assertThat(getResponses, not(hasItem(containDocument(FIRST_INDEX_NAME, FIRST_INDEX_ID_SONG_2))));
+    }
+
+    @Test
+    public void testSearchDocumentWithBoolOrTermDLSRestrictions() throws IOException, Exception {
+        SearchRequest searchRequest = new SearchRequest(FIRST_INDEX_NAME);
+
+        try (RestHighLevelClient restHighLevelClient = cluster.getRestHighLevelClient(USER_MATCH_ARTIST_BOOL_QUERY)) {
+            assertProperSearchResponseForAccessibleAndNonAccessibleDocuments(restHighLevelClient, searchRequest);
+        }
+
+        try (RestHighLevelClient restHighLevelClient = cluster.getRestHighLevelClient(USER_MATCH_STARS_TERM_QUERY)) {
+            assertProperSearchResponseForAccessibleAndNonAccessibleDocuments(restHighLevelClient, searchRequest);
+        }
+    }
+
+    @SuppressWarnings("unchecked")
+    private void assertProperSearchResponseForAccessibleAndNonAccessibleDocuments(
+        RestHighLevelClient restHighLevelClient,
+        SearchRequest searchRequest
+    ) throws IOException, Exception {
+        SearchResponse searchResponse = restHighLevelClient.search(searchRequest, DEFAULT);
+        assertThat(searchResponse, isSuccessfulSearchResponse());
+        assertThat(searchResponse, searchHitsContainDocumentsInAnyOrder(Pair.of(FIRST_INDEX_NAME, FIRST_INDEX_ID_SONG_1)));
+        assertThat(
+            searchResponse,
+            not(
+                searchHitsContainDocumentsInAnyOrder(
+                    FIRST_INDEX_SONGS_BY_ID.keySet()
+                        .stream()
+                        .filter(id -> !id.equals(FIRST_INDEX_ID_SONG_1))
+                        .map(id -> Pair.of(FIRST_INDEX_NAME, id))
+                        .collect(Collectors.toList())
+                )
+            )
+        );
```

Comment:
nit: test case on 720 and 721 can be combined to check that searchResponse only contains 1 entry and that is the song with id `FIRST_INDEX_ID_SONG_1`.